NbMGui/src/main/java/nbmgui/Controller.java

```diff
@@ -89,27 +134,35 @@ private void prepareObjectFile() {
         startButton.setDisable(false);
     }
 
-    private void fillProgramDataView(List<String> programData) {
-        VBox result = new VBox();
-        for (String lineStr : programData) {
-            HBox line = new HBox(new CheckBox(lineStr));
+    private void fillProgramDataView(List<String> programDataList) {
+        programData = new VBox();
+        for (String lineStr : programDataList) {
+            CheckBox line = new CheckBox(lineStr);
             line.setPadding(new Insets(1));
-            result.getChildren().add(line);
+            programData.getChildren().add(line);
         }
-        programDataView.setContent(result);
+        programDataView.setContent(programData);
     }
 
-    @FXML
-    void startProgram(ActionEvent event) {
-        Thread thread = new Thread(new Runnable() {
-            @Override
-            public void run() {
-                    machine.loadStringConstants(objectFile.getStringStorage());
-                    machine.loadProgram(0, objectFile.getProgram());
-                    machine.runProgram(0);
-                }
-        });
-        thread.start();
+    private void setCurrentPcOfUI() {
```

Comment:
I am a bit unclear what this method does. Is it highlighting the next instruction to be executed?